src/contracts/core/DelegationManager.sol

```diff
@@ -223,8 +217,12 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         eigenPodManager.forceIntoUndelegationLimbo(staker);
         
         // force a withdrawal of all of the staker's shares from the StrategyManager
-        return (strategyManager.forceTotalWithdrawal(staker));
+        bytes32 queuedWithdrawal = strategyManager.forceTotalWithdrawal(staker);
+
+        // actually undelegate the staker
+        _undelegate(staker);
```

Comment:
This method actually gets really funky if you think about it. `epm.forceIntoUndelegationLimbo` needs to query the staker's operator, and then also calls back into `DelegationManager.decreaseDelegatedShares(..., undelegateIfPossible: true)`

And this happens before we try to call the strategymanager, or even undelegate. I imagine there are similar weird loopy patterns there.

Reply:
addressed as part of PR https://github.com/Layr-Labs/eigenlayer-contracts/pull/183 -- or see the merge commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/176/commits/75e1c96437486846dce1b391fe59c71a88598113